drivers/hmis/spec/system/hmis/bulk_services_spec.rb

```diff
@@ -0,0 +1,107 @@
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+require 'rails_helper'
+require_relative '../../requests/hmis/login_and_permissions'
+require_relative '../../support/hmis_base_setup'
+
+RSpec.feature 'Hmis Form behavior', type: :system do
+  include_context 'hmis base setup'
+  include_context 'hmis service setup'
+
+  let!(:ds1) { create(:hmis_data_source, hmis: 'localhost') }
+
+  let!(:access_control) { create_access_control(hmis_user, p1) }
+  let(:bednight_service_type) { Hmis::Hud::CustomServiceType.find_by(hud_record_type: 200) }
+  let(:service_form_definition) { Hmis::Form::Definition.where(role: :SERVICE).first }
+  let!(:instance) { create(:hmis_form_instance, definition: service_form_definition, entity: p1, custom_service_type: bednight_service_type) }
+
+  let!(:pc1) { create :hmis_hud_project_coc, data_source: ds1, project: p1, coc_code: 'CO-500' }
+  let!(:c1) { create :hmis_hud_client, data_source: ds1, user: u1, first_name: 'Josiah' }
+  let!(:c2) { create :hmis_hud_client, data_source: ds1, user: u1, first_name: 'Joshua' }
+  let!(:c3) { create :hmis_hud_client, data_source: ds1, user: u1, first_name: 'Josephine' }
+
+  let(:today) { Date.current }
+
+  before(:each) do
+    sign_in(hmis_user)
+    visit "/projects/#{p1.id}"
+    click_link 'Bed Nights'
+
+    find('[name="search client"]').fill_in(with: 'Jos')
+    find_button('Search').trigger(:click)
+    expect(all('tbody tr').count).to eq(3) # all 3 clients returned because their names match the pattern
+  end
+
+  describe 'bed nights' do
+    it 'correctly searches, enrolls, and assigns service to clients' do
+      find('input[type="checkbox"][aria-label="select all"]', visible: :all).trigger(:click)
+      click_button 'Enroll (3) + Assign (3)'
+      assert_text 'Assigned' # wait for it to process
+
+      # Find the indices of the two columns we want to check
+      header_cells = all('thead th')
+      last_bed_night_date_index = header_cells.find_index { |cell| cell.text == 'Last Bed Night Date' }
+      assign_bed_night_index = header_cells.find_index { |cell| cell.text == "Assign Bed Night for #{today.strftime('%m/%d/%Y')}" }
+      expect(last_bed_night_date_index).not_to be_nil
+      expect(assign_bed_night_index).not_to be_nil
+
+      # Verify that all 3 rows have the expected attributes
+      all('tbody tr').each do |row|
+        # Check the "Last Bed Night Date" column
+        last_bed_night_date = row.all('td')[last_bed_night_date_index].text
+        expect(last_bed_night_date).to eq("Today (#{today.strftime('%m/%d/%Y')})")
+
+        # Check the "Assign Bed Night for mm/dd/yyyy" column
+        assign_button = row.all('td')[assign_bed_night_index].find('button')
+        expect(assign_button.text).to eq('Assigned')
+      end
+
+      services = Hmis::Hud::Service.joins(enrollment: [:project, :client]).all
```

Comment:
This feel unnecessarily prod and prone to interference, if there is another test that didn't clean up services. Can we just say `p1.services`?

Reply:
_And_ maybe wrapping the above `click_button` in an assertion to confirm that the block increases `(Hmis::Hud::Service, :count).by(3)` , so we feel confident that there were no services generated outside of the project.